frontend/src/app/Catalog/CatalogView/CatalogView.tsx

```diff
@@ -12,38 +12,86 @@ import { useHistory, useParams } from 'react-router';
 import { sortSections } from 'utils/sections/sort';
 import Skeleton from 'react-loading-skeleton';
 import ReadMore from './ReadMore';
-
+import { Tabs, Tab, Col } from 'react-bootstrap';
 import styles from './CatalogView.module.scss';
-import { useSelector } from 'react-redux';
+import { connect, useDispatch, useSelector, MapStateToProps } from 'react-redux';
 import SectionTable from './SectionTable';
+import { getLatestSemester, Semester } from 'utils/playlists/semesters';
+import EnrollmentGraphCard from 'components/GraphCard/EnrollmentGraphCard';
+import GradesGraphCard from 'components/GraphCard/GradesGraphCard';
+import { gradeReset, enrollReset, fetchGradeContext, fetchEnrollContext, fetchGradeFromUrl, fetchEnrollFromUrl, fetchGradeData, fetchEnrollData } from 'redux/actions';
+import GradesGraph from 'components/Graphs/GradesGraph';
+import axios from 'axios';
+import BTLoader from 'components/Common/BTLoader';
+import store from 'redux/store';
+import EnrollmentGraph from 'components/Graphs/EnrollmentGraph';
 
 interface CatalogViewProps {
 	coursePreview: CourseFragment | null;
 	setCurrentCourse: Dispatch<SetStateAction<CourseFragment | null>>;
 	setCurrentFilters: Dispatch<SetStateAction<CurrentFilters>>;
+	gradesData:any;
+	enrollmentData:any;
+	gradesGraphData:any;
+	enrollGraphData:any;
+	gradesSelectedCourses:any;
+	enrollSelectedCourses:any;
+	enrollContext:any;
 }
 
 const skeleton = [...Array(8).keys()];
 
+var dict = new Map([
+	['Field Work', 'FLD'],
+	['Session', 'SES'],
+	['Colloquium', 'COL'],
+	['Recitation', 'REC'],
+	['Internship', 'INT'],
+	['Studio', 'STD'],
+	['Demonstration', 'dem'],
+	['Web-based Discussion', 'WBD'],
+	['Discussion', 'DIS'],
+	['Tutorial', 'TUT'],
+	['Clinic', 'CLN'],
+	['Independent Study', 'IND'],
+	['Self-paced', 'SLF'],
+	['Seminar', 'SEM'],
+	['Lecture', 'LEC'],
+	['Web-based Lecture', 'WBL'],
+	['Web-Based Lecture', 'WBL'],
+	['Directed Group Study', 'GRP'],
+	['Laboratory', 'LAB'],
+  ]);
+
 const CatalogView = (props: CatalogViewProps) => {
-	const { coursePreview, setCurrentFilters, setCurrentCourse } = props;
+
+	const { coursePreview, setCurrentFilters, setCurrentCourse, gradesData, enrollmentData, gradesGraphData, enrollGraphData, gradesSelectedCourses, enrollSelectedCourses, enrollContext } = props;
 	const { abbreviation, courseNumber, semester } = useParams<{
 		abbreviation: string;
 		courseNumber: string;
 		semester: string;
 	}>();
 
+	const dispatch = useDispatch();
+
 	const [course, setCourse] = useState<CourseFragment | null>(coursePreview);
 	const [isOpen, setOpen] = useState(false);
+
 	const history = useHistory();
 
-	const legacyId = useSelector(
+	var legacyId = useSelector(
```

Comment:
the `legacyId` is never reassigned so should be `const` from what I can tell.

Reply:
fixed